src/wifi.rs

```diff
@@ -1355,3 +1358,110 @@ impl WifiWait {
         }
     }
 }
+
+#[cfg(all(feature = "async"))]
+pub struct EspAsyncWifiDriver<'d> {
+    sync: WifiDriver<'d>,
+}
+
+#[cfg(all(feature = "async"))]
+impl<'d> EspAsyncWifiDriver<'d> {
+    #[cfg(all(feature = "alloc", esp_idf_comp_nvs_flash_enabled))]
+    pub fn new<M: WifiModemPeripheral>(
```

Comment:
In terms of reusability, reimplementing everything which is currently in `WifiDriver` from scratch is not a great idea. What you might or might not have noticed (not really sure as the current code is just a bunch of stubs), is that the `WifiDriver` methods are in fact **non-blocking**, in that e.g. calling `start` does not mean that the driver would've started when the method returns. Quite the opposite - the driver would only *initiate* the start procedure, and then you have to subscribe to the event bus and wait for a "start" event notification. Ditto for `connect`.

Hence, what makes sense is to have your `AsyncWifiDriver` taking a `T: BorrowMut<WifiDriver>` parameter (as per above). All of your methods would just call the corresponding methods on `WifiDriver` and then asynchronously wait on the system event bus until the corresponding event is received. Also note that the system event bus (and all event buses in fact) have an async adaptor.

So you can possibly implement a helper struct - `AsyncWifiWait`  - similar in spirit to the blocking `WifiWait` - except that it would asynchronously wait. And then use it in your `AsyncWifiDriver.